pkg/bgpv1/gobgp/server.go

```diff
@@ -91,39 +92,72 @@ func NewGoBGPServerWithConfig(ctx context.Context, log *logrus.Entry, params typ
 // AddNeighbor will add the CiliumBGPNeighbor to the gobgp.BgpServer, creating
 // a BGP peering connection.
 func (g *GoBGPServer) AddNeighbor(ctx context.Context, n types.NeighborRequest) error {
+	peer, err := g.getPeerForNeighbor(n)
+	if err != nil {
+		return err
+	}
+	peerReq := &gobgp.AddPeerRequest{
+		Peer: peer,
+	}
+	if err = g.server.AddPeer(ctx, peerReq); err != nil {
+		return fmt.Errorf("failed while adding peer %v %v: %w", n.Neighbor.PeerAddress, n.Neighbor.PeerASN, err)
+	}
+	return nil
+}
+
+// UpdateNeighbor will update the existing CiliumBGPNeighbor in the gobgp.BgpServer.
+func (g *GoBGPServer) UpdateNeighbor(ctx context.Context, n types.NeighborRequest) error {
+	peer, err := g.getPeerForNeighbor(n)
+	if err != nil {
+		return err
+	}
+	peerReq := &gobgp.UpdatePeerRequest{
+		DoSoftResetIn: true, // should perform soft reset only if needed
```

Comment:
I did not see `DoSoftResetIn` being used anywhere in gobgp. For users, they cannot really do soft reset as we don't provide those hooks. I would assume session gets re-negotiated, but double check in container labs.

Reply:
You are right, it does not seem to be used anywhere in the GoBGP code.. I did not check, just trusted the documentation for the API :) Anyway, it is still being set by gobgp CLI on peer update, so I would keep it to be compliant the the docs and the CLI (e.g. for future changes). As you assumed, in reality it re-negotiates the session.